.github/workflows/editoast.yml

```diff
@@ -124,3 +124,25 @@ jobs:
         run: cargo doc --manifest-path editoast/Cargo.toml
         env:
           RUSTDOCFLAGS: "-D warnings" # Set warnings as errors
+
+  infra_schema-sync:
+    name: Check infra_schema.json
+    runs-on: ubuntu-latest
+    steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+          with:
+            # run the CI on the actual latest commit of the PR, not the attempted merge
+            ref: ${{ github.event.pull_request.head.sha }}
+        - name: Install poetry
+          run: pipx install poetry
+        - uses: actions/setup-python@v4
+          with:
+            python-version: "3.11"
+            cache: "poetry"
+        - name: Check infra_schema.json sync
+          run: |
+            cd python/osrd_schemas
+            poetry install --no-interaction --no-root
+            poetry run python osrd_schemas/infra.py > current_infra_schema.json
+            diff current_infra_schema.json ../../editoast/src/views/infra/infra_schema.json
```

Comment:
We should do this in `osrd_schema.yml`. So we check that only when the python schema changes.